### PR TITLE
Azure monitor/remove graf tracing

### LINKED ISSUE #73842: AzureMonitor: Remove infra tracing
Parent issue: https://github.com/grafana/grafana/issues/72265

### PR DESCRIPTION
**Pull request by bossinc**
https://github.com/grafana/grafana/issues/73842

We are removing grafana dependencies from Azure Monitor so we can eventually separate AM from core grafana.

This PR changes tracing to use the plugin-sdk package.

Notes for running Grafana tracing. 
Clone Tempo and run the docker compose in example/docker-compose/local. I changed the yaml to remove grafana and k6. There is probably a better way to do this but :shrug: 
Add  `[tracing.opentelemetry.otlp]` and `address=localhost:4317` to your ini
You might need to `export GF_DIAGNOSTICS_TRACING_ENABLED=true`
Run grafana
You should be able to make a tempo data source and see traces still work for AM

### CODE REVIEW

File: pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go [hunk 1]
```diff
@@ -253,14 +253,19 @@ func (e *AzureMonitorDatasource) retrieveSubscriptionDetails(cli *http.Client, c
...
 	req.URL.RawQuery = values.Encode()
 
-	ctx, span := tracer.Start(ctx, "azuremonitor query", trace.WithAttributes(
-		attribute.String("subscription", subscriptionId),
-		attribute.Int64("datasource_id", dsId),
-		attribute.Int64("org_id", orgId),
-	))
+	ctx, span := tracing.DefaultTracer().Start(
+		ctx,
+		"azuremonitor query",
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go:258]:
```suggestion
		"azuremonitor subscription query",
```

File: pkg/tsdb/azuremonitor/azuremonitor.go [hunk 2]
```diff
@@ -26,21 +25,19 @@ import (
...
 )
 
-func ProvideService(cfg *setting.Cfg, httpClientProvider *httpclient.Provider, features featuremgmt.FeatureToggles, tracer tracing.Tracer) *Service {
+func ProvideService(cfg *setting.Cfg, httpClientProvider *httpclient.Provider, features featuremgmt.FeatureToggles) *Service {
 	proxy := &httpServiceProxy{}
 	executors := map[string]azDatasourceExecutor{
 		azureMonitor:       &metrics.AzureMonitorDatasource{Proxy: proxy, Features: features},
 		azureLogAnalytics:  &loganalytics.AzureLogAnalyticsDatasource{Proxy: proxy},
 		azureResourceGraph: &resourcegraph.AzureResourceGraphDatasource{Proxy: proxy},
-		azureTraces:        &loganalytics.AzureLogAnalyticsDatasource{Proxy: proxy},
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/azuremonitor.go:35]: Why is this line removed?

**Review by aangelisc — APPROVED** [review]
LGTM, one suggestion and question in comments 👍

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go [hunk 3]
```diff
@@ -151,16 +150,17 @@ func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, query *
...
 		attribute.Int64("from", query.TimeRange.From.UnixNano()/int64(time.Millisecond)),
 		attribute.Int64("until", query.TimeRange.To.UnixNano()/int64(time.Millisecond)),
 		attribute.Int64("datasource_id", dsInfo.DatasourceID),
 		attribute.Int64("org_id", dsInfo.OrgID),
-	))
-	defer span.End()
+	),
+	)
 
-	tracer.Inject(ctx, req.Header, span)
```
Comment (marefr) [inline pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:163]: Note on removing this should be fine as long as you're using the httpclient package/provider/client (which I'm pretty sure you do here) to make HTTP requests since it does this automatically for you, see https://github.com/grafana/grafana/blob/1b4c2fc94892fb2aaefbc9f377b5f8cec22066d1/pkg/infra/httpclient/httpclientprovider/tracing_middleware.go#L37 and https://github.com/grafana/grafana-plugin-sdk-go/blob/dd41f8d709969a5efd5375833f9754cecf9176bc/backend/httpclient/tracing_middleware.go#L45

Just something to keep in mind since without this the receiving HTTP server won't receive trace id identifier in HTTP header and be able to add spans to an existing trace id - this obviously requires both grafana, the plugin and the HTTP server to send traces to the same storage.

**Review by marefr — APPROVED** [review]
LGTM with some nits/notes

File: pkg/tsdb/azuremonitor/azuremonitor.go (hunk 2 above)
Comment (bossinc) [reply pkg/tsdb/azuremonitor/azuremonitor.go:35]: I thought it was part of the old tracing package, but I see it's not. What/how is this used for? I can add it back

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go [hunk 4]
```diff
@@ -151,16 +150,17 @@ func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, query *
...
+	defer span.End()
+	backend.Logger.Debug("azure resource graph query", "traceID", trace.SpanContextFromContext(ctx).TraceID())
```
[the remark this answers is not in the document]
Comment (bossinc) [reply pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:163]: Good to know!
When there is no trace id it prints zeros for the trace id.

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go (hunk 3 above)
Comment (bossinc) [reply pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:163]: Thank you for the info!

File: pkg/tsdb/azuremonitor/azuremonitor.go (hunk 2 above)
Comment (bossinc) [reply pkg/tsdb/azuremonitor/azuremonitor.go:35]: omg omg omg I get it now lol
